src/lib/optimizer/chunk_statistics.hpp

```diff
@@ -0,0 +1,218 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+#include "storage/base_encoded_column.hpp"
+#include "storage/deprecated_dictionary_column.hpp"
+#include "storage/dictionary_column.hpp"
+#include "storage/reference_column.hpp"
+#include "storage/run_length_column.hpp"
+#include "storage/value_column.hpp"
+
+namespace opossum {
+
+class BaseFilter : public std::enable_shared_from_this<BaseFilter> {
```

Comment:
This class really needs doc

Reply:
Also, please move the classes to their own files. Right now, `chunk_encoder.cpp` includes the `RangeFilter`, which unnecessarily increases the compile time.